tests/unit/admin/test-class-wc-rest-payments-orders-controller.php

```diff
@@ -1605,6 +1605,116 @@ public function test_get_terminal_intent_payment_method_invalid_value() {
 		$this->controller->get_terminal_intent_payment_method( $request );
 	}
 
+	public function test_capture_terminal_payment_allows_charging_order_with_intent_meta() {
+		$order = $this->create_mock_order();
+		$order->update_meta_data( WC_Payments_Order_Service::INTENT_ID_META_KEY, $this->mock_intent_id );
+		$order->update_meta_data( WC_Payments_Order_Service::INTENTION_STATUS_META_KEY, Intent_Status::REQUIRES_CAPTURE );
+		$order->save_meta_data();
+
+		$mock_intent = WC_Helper_Intention::create_intention(
+			[
+				'status'   => Intent_Status::REQUIRES_CAPTURE,
+				'metadata' => [
+					'order_id' => $order->get_id(),
+				],
+			]
+		);
+
+		$request = $this->mock_wcpay_request( Get_Intention::class, 1, $this->mock_intent_id );
+
+		$request->expects( $this->once() )
+			->method( 'format_response' )
+			->willReturn( $mock_intent );
+
+		$this->mock_gateway
+			->expects( $this->once() )
+			->method( 'capture_charge' )
+			->with( $this->isInstanceOf( WC_Order::class ) )
+			->willReturn(
+				[
+					'status' => Intent_Status::SUCCEEDED,
+					'id'     => $this->mock_intent_id,
+				]
+			);
+
+		$this->order_service
+			->expects( $this->once() )
+			->method( 'attach_intent_info_to_order' )
+			->with(
+				$this->isInstanceOf( WC_Order::class ),
```

Comment:
It'd be easier to read if we used `$this->anything()` for parameters which are not essential to verify.

Reply:
Fixed in bab3118.